### PR TITLE
docs: add table calculations docs

### PR DESCRIPTION
**Pull request by TuringLovesDeathMetal**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
#504 

### Description:
Adds docs on table calculations + some table calculation templates that users can work from

### Scope of the changes (select all that apply):
- [ ] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [x] Docs  
- [ ] CI/CD

### CODE REVIEW

File: docs/docs/guides/adding-custom-metrics-table-calculations-explore.mdx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+- You can round your table calculation to only show a certain number of decimal points. Here's an example where I only show two decimal points:
+
+  ```sql
+  ROUND(${table.column_i_want_to_compare} * 100 / LAG(${table.column_i_want_to_compare}) OVER(ORDER BY ${table.column_I_want_to_order_by}, ${table.optional_other_column_I_want_to_order_by}), 2)
+  ```
+
+<img src={PercentOfTotalRounded} width="320" height="388" style={{display: "block", margin: "0 auto 20px auto"}}/>
+
+#### Percent change from previous
```
Comment (ZeRego) [inline docs/docs/guides/adding-custom-metrics-table-calculations-explore.mdx:185]: I find these titles small and hard to find in a wall of text 🔍 👀 Wondering if we should bump the template titles one level up.

**Review by ZeRego — APPROVED** [review]

File: docs/docs/guides/adding-custom-metrics-table-calculations-explore.mdx (hunk 1 above)
Comment (TuringLovesDeathMetal) [reply docs/docs/guides/adding-custom-metrics-table-calculations-explore.mdx]: yes! already did this too once I started looking at it :)

**Comment by hamzahc1** [comment]
@TuringLovesDeathMetal The content in these are great but I think they're too buried in the page right now because of the current page structure. 

Changes to structure: 
- Custom metrics should be it's own page (and appear in the sidebar)
- Table Calculations should be it's own page (and appear in the sidebar)
- The SQL templates should also exist in a sub-directory in the sidebar so you don't have to scroll through the whole page to find them (it's really hard to figure out what to look for right now and it's very busy). 

Changes to content: 
- Currently for each calculation template, there are two code blocks - we could make it easier for the user by putting above the relevant one "You can copy + paste the following SQL" or some obvious message that that's what they need to use to get the output. 
- Can we add line breaks and have the SQL word-wrap rather than single line scrolling? It would be a lot more legible that way. 

Otherwise, I think these are brilliant! We can share these links with folks individually then (and even link out to them in the app if needed)

**Review by hamzahc1 — CHANGES_REQUESTED** [review]
@TuringLovesDeathMetal The content in these are great but I think they're too buried in the page right now because of the current page structure. 

Changes to structure: 
- Custom metrics should be it's own page (and appear in the sidebar)
- Table Calculations should be it's own page (and appear in the sidebar)
- The SQL templates should also exist in a sub-directory in the sidebar so you don't have to scroll through the whole page to find them (it's really hard to figure out what to look for right now and it's very busy). 

Changes to content: 
- Currently for each calculation template, there are two code blocks - we could make it easier for the user by putting above the relevant one "You can copy + paste the following SQL" or some obvious message that that's what they need to use to get the output. 
- Can we add line breaks and have the SQL word-wrap rather than single line scrolling? It would be a lot more legible that way. 

Otherwise, I think these are brilliant! We can share these links with folks individually then (and even link out to them in the app if needed)

**Comment by TuringLovesDeathMetal** [comment]
@hamzahc1 I've added a bunch of the changes. Two things:

1. The only way to have SQL with line breaks in markdown is to use quotes. This changes the format a lot (here's what it looked like before for reference: 
image)
2. I can't figure out a way to anchor a heading to the sidebar. So either I create `table calculations` and `table calculation templates` as two separate markdown pages, and then link them both in a category. Or, we just leave it as-is.

**Comment by hamzahc1** [comment]
1. @TuringLovesDeathMetal - The changes to the code blocks now mean you can't copy + paste them easily and also don't look quite right:

Screenshot 2022-04-27 at 11 28 47

What I'm expecting to see is something more like a code block like this where the code doesn't scroll, they've used line breaks to make it word-wrap instead: https://docusaurus.io/docs/markdown-features/code-blocks

2. In that case - all the sections up until "📝 Table calculation SQL templates" can stay where they are under the current heading (Adding table calc...) and at the end of that section we can link to the SQL template section. And we should create a separate markdown category dropdown which would reveal each SQL template as a separate page. The SQL templates are such a valuable resource they should be highlighted separately.

**Comment by TuringLovesDeathMetal** [comment]
@hamzahc1 ya, their code blocks have new lines because the code they're using has new lines...unfortunately SQL `OVER` clauses are all just one line of code :( I'll see what I can do with trying to split it up across lines, but personally, I find SQL harder to read when it's split across lines 

> And we should create a separate markdown category dropdown which would reveal each SQL template as a separate page. The SQL templates are such a valuable resource they should be highlighted separately.

I feel like I'd want to see all of the available SQL templates on one page, no? The functions aren't so big/dissimilar that I'd want to look at them as separate pages. Happy to split them from `table calculation` general info, but I was thinking of formatting it more like this:

- table calculations
  - Using table calculations in your Explore
  - Table calculation templates

**Review by hamzahc1 — APPROVED** [review]

**Pull request merged**